js_modules/dagit/packages/core/src/workspace/VirtualizedAssetTable.tsx

```diff
@@ -0,0 +1,376 @@
+        assetsForGroup.forEach(({id, assetKey}) => {
+          flat.push({type: 'asset', id, path: assetKey.path});
+        });
+      }
+    });
+    return flat;
+  }, [grouped, expandedKeys]);
+
+  const rowVirtualizer = useVirtualizer({
+    count: flattened.length,
+    getScrollElement: () => parentRef.current,
+    estimateSize: (ii: number) => {
+      const row = flattened[ii];
+      return row?.type === 'group' ? 48 : 64;
+    },
+    overscan: 10,
+  });
+
+  const totalHeight = rowVirtualizer.getTotalSize();
+  const items = rowVirtualizer.getVirtualItems();
+
+  return (
+    <>
+      <Box
+        border={{side: 'horizontal', width: 1, color: Colors.KeylineGray}}
+        style={{
+          display: 'grid',
+          gridTemplateColumns: '40% 30% 20% 10%',
+          height: '32px',
+          fontSize: '12px',
+          color: Colors.Gray600,
+        }}
+      >
+        <HeaderCell>Asset name</HeaderCell>
+        <HeaderCell>Materialized</HeaderCell>
+        <HeaderCell>Latest run</HeaderCell>
+        <HeaderCell>Actions</HeaderCell>
+      </Box>
+      <div style={{overflow: 'hidden'}}>
+        <Container ref={parentRef}>
+          <Inner $totalHeight={totalHeight}>
+            {items.map(({index, key, size, start}) => {
+              const row: RowType = flattened[index];
+              const type = row!.type;
+              return type === 'group' ? (
+                <GroupNameRow
+                  repoAddress={repoAddress}
+                  groupName={row.name}
+                  assetCount={row.assetCount}
+                  expanded={expandedKeys.includes(row.name)}
+                  key={key}
+                  height={size}
+                  start={start}
+                  onToggle={onToggle}
+                />
+              ) : (
+                <AssetRow
+                  key={key}
+                  path={row.path}
+                  repoAddress={repoAddress}
+                  height={size}
+                  start={start}
+                />
+              );
+            })}
+          </Inner>
+        </Container>
+      </div>
+    </>
+  );
+};
+
+interface JobRowProps {
+  path: string[];
+  repoAddress: RepoAddress;
+  height: number;
+  start: number;
+}
+
+const AssetRow = (props: JobRowProps) => {
+  const {path, repoAddress, start, height} = props;
+
+  const [queryAsset, queryResult] = useLazyQuery<SingleAssetQuery, SingleAssetQueryVariables>(
+    SINGLE_ASSET_QUERY,
+    {
+      fetchPolicy: 'cache-and-network',
+      variables: {input: {path}},
+    },
+  );
+
+  useDelayedRowQuery(queryAsset);
+  const {data} = queryResult;
+
+  const asset = React.useMemo(() => {
+    if (data?.assetOrError.__typename === 'Asset') {
+      return data.assetOrError;
+    }
+    return null;
+  }, [data]);
+
+  const assetKeyAsString = JSON.stringify(path);
+  const liveData = React.useMemo(() => {
+    if (asset?.definition && data?.assetsLatestInfo) {
```

Comment:
I might be misunderstanding some logic inside useDelayedRowQuery, but I think the .find in here could probably be a `[0]` since we are asking for each asset individually in it's own query now? This is still more robust but we could probably skip the path stringifying if we wanted to?